server/src/test/java/io/druid/server/coordinator/LoadQueuePeonTest.java

```diff
@@ -0,0 +1,393 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.server.coordinator;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Function;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import io.druid.concurrent.Execs;
+import io.druid.curator.CuratorTestBase;
+import io.druid.jackson.DefaultObjectMapper;
+import io.druid.server.coordination.DataSegmentChangeCallback;
+import io.druid.server.coordination.DataSegmentChangeHandler;
+import io.druid.server.coordination.DataSegmentChangeRequest;
+import io.druid.server.coordination.SegmentChangeRequestDrop;
+import io.druid.server.coordination.SegmentChangeRequestLoad;
+import io.druid.timeline.DataSegment;
+import io.druid.timeline.partition.NoneShardSpec;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.recipes.cache.PathChildrenCache;
+import org.apache.curator.framework.recipes.cache.PathChildrenCacheEvent;
+import org.apache.curator.framework.recipes.cache.PathChildrenCacheListener;
+import org.apache.curator.utils.ZKPaths;
+import org.joda.time.Duration;
+import org.joda.time.Interval;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class LoadQueuePeonTest extends CuratorTestBase
+{
+  private static final String LOAD_QUEUE_PATH = "/druid/loadqueue/localhost:1234";
+
+  private final ObjectMapper jsonMapper = new DefaultObjectMapper();
+
+  private LoadQueuePeon loadQueuePeon;
+  private PathChildrenCache loadQueueCache;
+
+  @Before
+  public void setUp() throws Exception
+  {
+    setupServerAndCurator();
+    curator.start();
+    curator.create().creatingParentsIfNeeded().forPath(LOAD_QUEUE_PATH);
+
+    loadQueueCache = new PathChildrenCache(
+        curator,
+        LOAD_QUEUE_PATH,
+        true,
+        true,
+        Execs.singleThreaded("load_queue_cache-%d")
+    );
+  }
+
+  @Test
+  public void testMultipleLoadDropSegments() throws Exception
+  {
+    final AtomicInteger requestSignalIdx = new AtomicInteger(0);
+    final AtomicInteger segmentSignalIdx = new AtomicInteger(0);
+
+    loadQueuePeon = new LoadQueuePeon(
+        curator,
+        LOAD_QUEUE_PATH,
+        jsonMapper,
+        Execs.scheduledSingleThreaded("test_load_queue_peon_scheduled-%d"),
+        Execs.singleThreaded("test_load_queue_peon-%d"),
+        new DruidCoordinatorConfig()
```

Comment:
there are multiple places where we create DruidCoordinatorConfig, 
can we create TestDruidCoordinatorConfig and use that in the tests instead of creating multiple anonymous classes.